pkg/sql/parser/sql.y

```diff
@@ -6510,15 +6507,15 @@ opt_using_clause:
 db_level_changefeed_filter_option:
   EXCLUDE TABLES table_name_list
   {
-    $$.val = &tree.ChangefeedFilterOption{Tables: $3.tableNames(), FilterType: tree.ExcludeFilter}
+    $$.val = tree.ChangefeedFilterOption{Tables: $3.tableNames(), FilterType: tree.ExcludeFilter}
   }
 | INCLUDE TABLES table_name_list
   {
-    $$.val = &tree.ChangefeedFilterOption{Tables: $3.tableNames(), FilterType: tree.IncludeFilter}
+    $$.val = tree.ChangefeedFilterOption{Tables: $3.tableNames(), FilterType: tree.IncludeFilter}
   }
 | /* EMPTY */ 
   {
-    $$.val = nil
+    $$.val = tree.ChangefeedFilterOption{FilterType: tree.ExcludeFilter}
```

Comment:
nit: as mentioned in https://github.com/cockroachdb/cockroach/pull/154187#discussion_r2389576087, I'd still prefer to see this just be the zero value since the zero value of the enum is already `ExcludeFilter`.

in go, it's a best practice to make sure the zero value is useful (which it already is in this case)

```suggestion
    $$.val = tree.ChangefeedFilterOption{}
```